src/NuGet.Core/NuGet.Packaging.Core/PackageType.cs

```diff
@@ -87,5 +89,28 @@ public override int GetHashCode()
 
             return combiner.CombinedHash;
         }
+
+        public int CompareTo(PackageType other)
+        {
+
+            if (other == null)
+            {
+                return 1;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return 0;
+            }
+
+            var res = StringComparer.OrdinalIgnoreCase.Compare(Name, other.Name);
+
+            if(res != 0)
+            {
+                return res;
+            }
+
+            return Version.CompareTo(other.Version);
```

Comment:
Looks like Version could be null here and throw. Maybe the parser prevents this, but this class seems to allow it.

Reply:
There's a null version check, so version can not be null.